Samples/PDESampleApplication/Form2.cs

```diff
@@ -0,0 +1,319 @@
+                {
+                    g_selectedFile = selectedPath;
+                } 
+                else if (Directory.Exists(selectedPath))
+                {
+                    g_selectedFolder = selectedPath; 
+                } else
+                {
+                    MessageBox.Show("Selected path is neither a file nor a directory please rectify");
+                }
+            } 
+        }
+
+        private void button5_Click(object sender, EventArgs e)
+        {
+            if (openFileDialog1.ShowDialog() == DialogResult.OK)
+            {
+                listView2.Items.Add(openFileDialog1.FileName.Trim());
+                g_selectedFile = openFileDialog1.FileName;
+                if (!File.Exists(g_selectedFile))
+                {
+                    MessageBox.Show("Selected file is not a file. please rectify");
+                }
+            }
+        }
+
+
+        private async void FolderL1_Click(object sender, EventArgs e)
+        {
+            g_udpm = UserDataProtectionManager.TryGetDefault();
+            if (g_udpm == null)
+            {
+                LogLine("Personal Data Encryption is not supported or enabled. Restart this app to check again.");
+                return;
+            }
+            if (g_selectedFolder.Length > 0)
+            {
+                StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(g_selectedFolder);
+                this.ProtectFolderRecursively(folder, UserDataAvailability.AfterFirstUnlock);
+            }
+        }
+
+        private async void FolderL2_Click(object sender, EventArgs e)
+        {
+            g_udpm = UserDataProtectionManager.TryGetDefault();
+            if (g_udpm == null)
+            {
+                LogLine("Personal Data Encryption is not supported or enabled. Restart this app to check again.");
+                return;
+            }
+            if (g_selectedFolder.Length > 0)
+            {
+                StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(g_selectedFolder);
+                this.ProtectFolderRecursively(folder, UserDataAvailability.WhileUnlocked);
+            }
+        }
+
+        private async void FolderUnprotect_Click(object sender, EventArgs e)
+        {
+            g_udpm = UserDataProtectionManager.TryGetDefault();
+            if (g_udpm == null)
+            {
+                LogLine("Personal Data Encryption is not supported or enabled. Restart this app to check again.");
+                return;
+            }
+            if (g_selectedFolder.Length > 0)
+            {
+                StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(g_selectedFolder);
+                this.ProtectFolderRecursively(folder, UserDataAvailability.Always);
+            }
+        }
+
+        private async void FileL1_Click(object sender, EventArgs e)
+        {
+            var udpm = UserDataProtectionManager.TryGetDefault();
```

Comment:
Inconsistent use of udpm and g_udpm. It's ok to always instantiate UserDataProtectionManager, it's a lightweight call.